Unexpected Rust ownership transfer in NyxHelper.set_timeout method

**Bug description**
When the method below is called, the `NyxHelper` object will be passed in as `mut self` and an unexpected ownership transfer will happen. After the method ends, the local variable `self` will be destroyed and the `NyxHelper` object cannot be used anymore.
https://github.com/AFLplusplus/LibAFL/blob/0c9933c3e4e91f1ada05b02dd2c38e9dd47ffe3f/libafl_nyx/src/helper.rs#L136

**To Reproduce**
Code to reproduce the behavior:
```
let mut helper = NyxHelper::new(share_dir, cpu_id, true, false, None).unwrap();
helper.set_timeout(Duration::from_millis(100));
let map_observer = unsafe { StdMapObserver::from_mut_ptr("trace", helper.trace_bits, helper.map_size) }; // Rust compiler will report error here.
```

**Screen output**
```
error[E0382]: use of moved value: `helper`
  --> src/main.rs:25:90
   |
22 |     let mut helper = NyxHelper::new(share_dir, cpu_id, true, false, None).unwrap();
   |         ---------- move occurs because `helper` has type `NyxHelper`, which does not implement the `Copy` trait
23 |     // set helper's timeout to 0.1 seconds
24 |     helper.set_timeout(Duration::from_millis(100));
   |     ------ value moved here
25 |     let map_observer = unsafe { StdMapObserver::from_mut_ptr("trace", helper.trace_bits, helper.map_size) };
   |                                                                                          ^^^^^^^^^^^^^^^ value used here after move
```
**How to fix**
I think you just need to add an "&" before `mut self`, so that self is a reference of the object.
`pub fn set_timeout(&mut self, time: Duration) {`

--- a/libafl_nyx/src/helper.rs
+++ b/libafl_nyx/src/helper.rs
@@ -133,7 +133,7 @@ impl NyxHelper {
     }
 
     /// Set a timeout for Nyx
-    pub fn set_timeout(mut self, time: Duration) {
+    pub fn set_timeout(&mut self, time: Duration) {
         let sec: u8 = time
             .as_secs()
             .try_into()

